cfi_backward.adoc

```diff
@@ -128,98 +148,157 @@ instructions on such machines is not supported.
+is as follows:
+
+[literal]
+    function_entry:
+        sspush x1      # push link register x1 on shadow stack
+         :
+         : 
+        sspop  x1      # pop return address from shadow stack
+        ret
+
+These examples illustrates the use of `x1` register by the ABI as the link
+register. Alternatively, the ABI may use `x5` as the link register.
 
 A leaf function i.e. a function that does not itself make function calls does
 not need to push the link register to the shadow stack or pop it from the shadow
-stack. The return value may be held in the link register itself for the duration
-of the function execution.
-
-In the RVI psABI, the x1 register is designated as the link register and x5
-register is designated as a temporary register. Since the shadow link register
-is loaded at the tail of the function, prior to return, the x5 register being
-used as the shadow link register does not impose a burden on the compiler as the
-x5 register, being a temporary register that is not preserved across a call, is
-usually free for use at the tail of the function.
-
-Programs operating in the control stack mode may store the return address only
-to the shadow stack in the function prologue. Such functions when returning from
-the function load the link register value from the shadow stack. Such programs
-may either use the x1 or x5 register, depending on their ABI, as the link
-register. The hardware return-address prediction stacks detect the use of x1/x5
-as the rd and x1/x5 as the source for a JALR instruction to infer if the JALR is
-used for a call or a return semantic for the purposes of prediction. To support
-both options the CFI extension provides the x1 and x5 variants of the shadow
-stack load and store.
+stack in either shadow stack mode or in control stack mode. The return value may
+be held in the link register itself for the duration of the leaf function
+execution.
 ====
 
-`sspop` performs a load identically to the existing `LOAD` instruction with the
-difference that the base is implicitly `ssp`, the width is implicitly `XLEN`.
-`sspush` performs a store identically to the existing `STORE` instruction with
-the difference that the base is implicitly `ssp`, the width is implicitly `XLEN`.
+The `sspop` and `sspopchkra` instructions perform a load identically to the
+existing `LOAD` instruction with the difference that the base is implicitly
+`ssp`, the width is implicitly `XLEN`. 
 
-The `sspush` and `sspop` require the virtual address in `ssp` to have a shadow stack
-attribute (see Shadow Stack Memory Protection). 
+The `sspush` instruction performs a store identically to the existing `STORE`
+instruction with the difference that the base is implicitly `ssp`, the width
+is implicitly `XLEN`.
 
-If the virtual address in `ssp` is not `XLEN` aligned then the instructions cause a
-load or store/AMO address-misaligned exception.
+The `sspush`, `sspopchkra`, and `sspop` require the virtual address in `ssp` to
```

Comment:
nit:  If virtual memory enabled, the `sspush`, `sspopchkra`, and `sspop` require the virtual address in `ssp` ....